crates/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -916,8 +916,14 @@ impl<'interner> TypeChecker<'interner> {
                         span,
                     });
                 }
-                let comptime = comptime_x.and(comptime_y, op.location.span);
-                Ok(Integer(comptime, *sign_x, *bit_width_x))
+                if op.is_bit_shift()
+                    && (*sign_x != Signedness::Unsigned || *sign_y != Signedness::Unsigned)
```

Comment:
This check won't catch the case where either integer type is inferred. You'll probably need to add something to the delayed type check on line 876

Reply:
I added an additional check there.